repos/system_upgrade/common/actors/scansourcefiles/tests/unit_test_scansourcefiles.py

```diff
@@ -1,5 +1,130 @@
-def test_scansourcefiles():
-    # TODO(pstodulk): keeping unit tests for later after I check the idea
-    # of this actor with the team.
-    # JIRA: OAMG-10367
-    pass
+import os
+
+import pytest
+
+from leapp.libraries.actor import scansourcefiles
+from leapp.libraries.common import testutils
+from leapp.libraries.stdlib import api
+from leapp.models import FileInfo, TrackedFilesInfoSource
+
+CUR_DIR = os.path.dirname(os.path.abspath(__file__))
+FILES_DIR = os.path.join(CUR_DIR, 'files')
+
+# values in list should correspond with the return values of get_source_major_version()
+RHEL_MAJOR_VERSIONS_LIST = list(range(8, 9+1))
+
+TRACKED_FILES_BY_VERSIONS = {
+    version: ['{}/file_major_version_{}'.format(FILES_DIR, version)]
```

Comment:
I was using `os.path.join` from start, but thought It would probably be more slower (because code uses many instances of it). But you are right, it is overally better, readibility, cross-platform ...

Reply:
I think it would require a lot more calls to notice any performance hit, anyway the test already run for quite a while and in this case correctness > performance.

Reply:
@tomasfratrik I recommend you to watch the following video to understand when it's good and when it's premature to optimise something: https://youtu.be/tKbV6BpH-C8 

In short, optimisation that is faster by 0.01s does not make sense when the execution time is in minutes.